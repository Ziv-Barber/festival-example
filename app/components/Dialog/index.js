// @flow

import React from 'react' // eslint-disable-line no-unused-vars

// Other components that we need:

// Internationalization (i18n):
// import { FormattedMessage } from 'react-intl'
// import createMessages from './messages'

// Material-ui stuff:
import MuiDialog from '@material-ui/core/Dialog'
import MuiDialogTitle from '@material-ui/core/DialogTitle'
import MuiDialogContent from '@material-ui/core/DialogContent'
import MuiDialogContentText from '@material-ui/core/DialogContentText'
import MuiDialogActions from '@material-ui/core/DialogActions'
import MuiSlide from '@material-ui/core/Slide'
import MuiButton from '@material-ui/core/Button'

// Our types:
import type { DialogPropsType } from './index.types'

const Transition = (props) => <MuiSlide direction="up" {...props} />

/**
 * Dialog react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const Dialog = (props: DialogPropsType) => {
  const { onClose, ...other } = props

  const handleClose = () => {
    onClose()
  }

  return (
    <MuiDialog
      onClose={handleClose}
      className={props.className || 'popup'}
      TransitionComponent={Transition}
      keepMounted
      aria-labelledby="popup-dialog-title"
      aria-describedby="popup-dialog-description"
      {...other}
    >
      <MuiDialogTitle id="popup-dialog-title">{props.title}</MuiDialogTitle>
      <MuiDialogContent>
        <MuiDialogContentText id="popup-dialog-description">
          {props.children}
        </MuiDialogContentText>
      </MuiDialogContent>
      <MuiDialogActions>
        <MuiButton onClick={handleClose} color="primary">
          OK
        </MuiButton>
      </MuiDialogActions>
    </MuiDialog>
  )
}

export default Dialog

// In case that someone need our types:
export type { DialogPropsType as Props }
