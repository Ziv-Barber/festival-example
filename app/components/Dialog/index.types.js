// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars

export type DialogPropsType = {
  /** Is true to show the dialog, else you'll see noting. */
  open: boolean,
  /** Close handler. */
  onClose: () => void,
  /** Styles for the dialog. */
  className?: string,
  /** Title for the dialog. */
  title: string,
  /** The children will be placed inside the dialog content. */
  children?: React.Node
}
