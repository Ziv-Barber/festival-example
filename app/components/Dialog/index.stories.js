import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import Dialog from './index'

storiesOf('Dialog', module).add(
  'basic',
  () => <Dialog text="Basic example" />,
  { notes: { markdown: 'This is an example to the Dialog component.' } }
)
