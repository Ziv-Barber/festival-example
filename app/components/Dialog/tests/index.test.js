// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import Dialog from '../index'

// For flow:
let { describe, it, afterEach, expect } = global

const handleClose = () => {}

/** @test {Dialog} */
describe('<Dialog />', () => {
  afterEach(cleanup)

  /** @test {Dialog} */
  it('should match snapshort', () => {
    const { container } = render(
      <Dialog open={true} onClose={handleClose} title="My Dialog" />
    )
    expect(container).toMatchSnapshot()
  })

  /** @test {Dialog} */
  it('should render', () => {
    const { getByText } = render(
      <Dialog open={true} onClose={handleClose} title="My Dialog" />
    )
    expect(getByText(/^My Dialog/)).not.toBeNull()
  })
})
