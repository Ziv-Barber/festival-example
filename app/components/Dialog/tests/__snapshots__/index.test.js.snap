// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should match snapshort 1`] = `
<div
  aria-hidden="true"
/>
`;
